spec/index.bs

```diff
@@ -13599,12 +13599,22 @@ enum GPUCanvasAlphaMode {
     "premultiplied",
 };
 
+enum GPUCanvasColorMode {
+    "default",
+    "extended",
```

Comment:
Maybe we can think of more descriptive names for these. "default" doesn't need to be called that because it _is_ the default, instead it can describe its behavior. `"clamp-to-sdr"`?

Similarly "extended" doesn't really distinguish it from any future modes (with different tonemapping?). Maybe `"clamp-to-display"` or `"clamp-to-screen"`?

Reply:
I like this idea for the names, if `extended` will clamp to the display's screen then it is not really extended on some older displays

Reply:
For `"default"`, I was thinking to call this `"standard"`, because this is in effect saying "the content will be clamped to the standard dynamic range of the monitor" but with less verbosity.

The `"extended"` term is becoming something of a term of art for "a space that goes above 1.0". It also matches, e.g, the [CAMetalLayer wantesExtendedDynamicRangeContent API](https://developer.apple.com/documentation/quartzcore/cametallayer/1478161-wantsextendeddynamicrangecontent). So I do think there's value in that particular terminology.

So with that in mind, does `"standard"`, `"extended"` sound good, or are there other things you'd like to go for.

Reply:
If we predict the possible existence of non-clamping (rolloff) modes for either sdr or extended, I'd prefer to have "clamp" in the names so they won't be ambiguous in the future. Other than that I don't think they need to be super long.

Also if there's no verb (like if it's just "standard"/"extended") then the type and member names `GPUCanvasColorMode` and `mode` should ideally be more descriptive, e.g. simply `GPUCanvasDynamicRange` `dynamicRange`

Reply:
Unless we predict other modes that aren't strictly dynamic range options would get mixed in here.

Reply:
> That said if there's no verb (like if it's just "standard"/"extended") then the type and member names GPUCanvasColorMode and mode should ideally be more descriptive

Thanks, I think that gets to the core of the issue. This `mode` variable is specifying "what is the tone mapping mode" (and nothing else). The potential other metadata that can be added is "metadata that the specified tone mapping mode will use".

And so to that end, I think I should rename `GPUCanvasColorMetadata` to `GPUCanvasToneMapping`, and `GPUCanvasColorMode` to `GPUCanvasToneMappingMode`. And the verbs make more sense. (Maybe I had anchored myself to shape of the `CAEDRMetadata` API... I'll blame that).

I've updated the patch to reflect this (and I think it's much more clear now!).

So, in this case it would be:
```
  context.configure({ toneMapping: { mode:"clamp-to-standard" } ... });
  context.configure({ toneMapping: { mode:"clamp-to-extended" } ... });
```
And the reason for using a dictionary instead of just an enum is that we're very likely going to have modes that allow parameters be requested.

E.g, for HDR10 (if we decide to go that route):
```
  context.configure({ toneMapping: { mode:"hdr10", clli:{maxFALL:200, maxCLL:1200} } ... });
```

I'm also hoping that the industry (images, video, and games) lands on a standardized vocabulary for custom tone mapping (the usual 1D curves(s) followed by a 3D LUT), which we can put here. That scheme will need quite a bit of ironing out (to ensure that we align with the various standards, and also to enable potential optimizations).

Reply:
Looks great, I think the names are very clear!